OnboardProject/src/Pages/AddUsers/AddUsersPage.tsx

```diff
@@ -0,0 +1,159 @@
+import { View, Text, TextInput, TouchableOpacity, Button } from 'react-native';
+import React from 'react';
+import { styles, getFormButtonTextAndColor } from '../GlobalStyles/GlobalStyles';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import { retrieveData } from '../../Utils/LocalStorage';
+import { clientWithAuth } from '../../Utils/GQL/clients';
+//import { usersQuery } from '../../Utils/GQL/clients';
+import { UserType } from '../../Utils/GQL/types';
+//import { styles } from './UsersStyles';
+
+interface AddUsersPageProps { }
+interface AddUsersPageState { 
+    user: UserType,
+    show: boolean,
+    date: Date,
+    isLoading:boolean,
+    errorMessage: string
+}
+
+class AddUsersPage extends React.Component<AddUsersPageProps, AddUsersPageState>{
+
+    constructor(props: AddUsersPageProps) {
+        super(props);
+        this.state = {
+            user:{id:'',name:'',phone:'',email:'',role:'',birthDate:'',__typename:''},
+            show:false,
+            date: new Date(),
+            isLoading:false,
+            errorMessage:""
+        }
+    }
+
+    handleAddUser = () => {
+
+    }
+    showDatePicker = () => {
+        this.setState({show:true});
+    }
+
+    onDateChange = (event: Event, selectedDate?:Date|undefined) => {
+        const currentDate = selectedDate || this.state.date;
+        this.setState({date:currentDate});
+        this.setState({show:false});
+    };
+
+    onNameChange = (value:string) => 
+        this.setState((prevState) => {
+            let user = {
+                ...prevState.user,
+                name: value
+            }
+            return {user};
+        });
+    onEmailChange = (value:string) => 
+        this.setState((prevState) => {
+            let user = {
+                ...prevState.user,
+                email: value
+            }
+            return {user};
+        });
+    onPasswordChange = (value:string) => 
+        this.setState((prevState) => {
+            let user = {
+                ...prevState.user,
+                email: value
+            }
+            return {user};
+        });
+    onPhoneChange = (value:string) => 
+        this.setState((prevState) => {
+            let user = {
+                ...prevState.user,
+                phone: value
+            }
+            return {user};
+        });
+    onRolechange = (value:string) => 
+        this.setState((prevState) => {
+            let user = {
+                ...prevState.user,
+                role: value
+            }
```

Comment:
Isso vale pra todos os seus setStates anteriores: se você fizer apenas `this.setState({ role: value })`, ele vai mudar apenas a role, e já deixar o restante da forma que estava. Não precisa pegar o estado anterior no callback e setar novamente aqui

Reply:
Na verdade, o state da página tem um campo user, que é do tipo UserType (definido em Utils/GQL/types), aí o TS não me deixa settar as propriedades do usuário direto, ele diz que eu tenho que escolher um dos states (no caso eu atualizo sempre o user).

Reply:
Ah, verdade, achei q tava direto no state os campos.